packages/mdc-textfield/foundation.js

```diff
@@ -190,6 +194,27 @@ class MDCTextfieldFoundation extends MDCFoundation {
   }
 
   /**
+   * Changes the floating label's state
+   * @private
+   */
+  updateLabelFloat_() {
+    const input = this.getNativeInput_();
+    const {LABEL_FLOAT_ABOVE} = MDCTextfieldFoundation.cssClasses;
+
+    if (this.isFocused_ || input.value) {
+      if (!this.labelFloatIsActive_) {
+        this.labelFloatIsActive_ = true;
+        this.adapter_.addClassToLabel(LABEL_FLOAT_ABOVE);
+      }
+    } else {
+      if (this.labelFloatIsActive_) {
```

Comment:
you actually don't even need this check. It doesnt matter what the internal state of this.labelFloatIsActive is....if neither `this.isFocused_` or `input.value` is true, then you should remove LABEL_FLOAT_ABOVE from the label element.